### DISCUSSION TITLE
Get Rollback Reason Within `after_rollback`

### DISCUSSION THREAD

**Discussion by jvanderen1** [jeremyevans/sequel#2231] [General]
I was wondering, is there a way to extract why a transaction rolled back within the `after_rollback` hook?

i.e.
```ruby
DB.transaction do
 DB.after_rollback do
    # Why did this rollback?
 end

 # ...
end
```

**Comment by jeremyevans** [comment]
Not currently.  What would you like to see there?  Potentially, we could make it possible to access the exception raised if the exception caused the rollback, or whether `rollback_on_exit` was called.

**Reply by jvanderen1** [reply]
I feel like it would be helpful to access the `Sequel::Rollback` error raised. And this could include the original wrapped exception or one that the database itself reports (in our case, a PostgreSQL error).

It would be helpful for us in debugging why a certain transaction gets rolled back on us 😃

**Reply by jeremyevans** [reply]
Running with a Database logger that logs at error level should make it possible to see if there was an exception raised by the database that may have caused the error.  It won't handle manual `Sequel::Rollback` raises, though, or calls to `rollback_on_exit`.

I think this would be a useful feature, so I'm open for including it in Sequel.  Is this something you would like to give a shot at implementing?

**Reply by jvanderen1** [reply]
I could try and give it a shot on my spare time 👍🏻
